src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -152,9 +178,9 @@ private void Execute(Transaction transaction, BlockHeader block, ITxTracer txTra
                 else
                 {
                     TraceLogInvalidTx(transaction, $"SENDER_ACCOUNT_DOES_NOT_EXIST {caller}");
-                    if (isCall || gasPrice == UInt256.Zero)
+                    if (!commit || restore || gasPrice == UInt256.Zero)
```

Comment:
why !commit || restore? what are other options? maybe keep this switch separate from others?

Reply:
there is test for that